SpokestackTests/AudioControllerTest.swift

```diff
@@ -14,59 +14,117 @@ import AVFoundation
-        let processFrameExpectation = expectation(description: "testStartStreaming calls AudioControllerTestDelegate as the result of processFrame method completion")
+        controller.configuration = config
+        controller.context = context
+        let delegate = AudioControllerTestDelegate(config, context: context)
+        context.listeners = [delegate]
+        context.stageInstances = [delegate]
+        let setupFailedExpectation = expectation(description: "testStartStreaming calls AudioControllerTestDelegate as the result of failure method completion")
 
         // Uninititalized delegates do not cause an exception during startStreaming
-        XCTAssertNoThrow(controller.startStreaming(context: context))
+        XCTAssertNoThrow(try AVAudioSession.sharedInstance().setCategory(.playAndRecord))
+        controller.startStreaming()
+        XCTAssertFalse(delegate.didSetupFailed)
+        
         /// stopStreaming does not cause an exception
-        XCTAssertNoThrow(controller.stopStreaming(context: context))
+        controller.stopStreaming()
+        XCTAssertFalse(delegate.didSetupFailed)
         
-        // Initialized PipelineDelegate is called during startStreaming
-        controller.pipelineDelegate = delegate
-        controller.delegate = delegate
+        // Incompatible AVAudioSession category fails
+        delegate.reset()
+        context.listeners = [delegate]
+        context.stageInstances = [delegate]
+        controller.context = context
         delegate.asyncExpectation = setupFailedExpectation
         XCTAssertNoThrow(try AVAudioSession.sharedInstance().setCategory(.ambient))
-        XCTAssertNoThrow(controller.startStreaming(context: context))
+        controller.startStreaming()
         wait(for: [setupFailedExpectation], timeout: 1)
         XCTAssert(delegate.didSetupFailed)
+        
         /// stopStreaming does not cause an exception
-        XCTAssertNoThrow(controller.stopStreaming(context: context))
-
-        delegate.reset()
+        controller.stopStreaming()
+    }
+    
+    func testProcess() {
+        let controller = AudioController.sharedInstance
+        let config = SpeechConfiguration()
+        let context = SpeechContext()
+        controller.configuration = config
+        controller.context = context
+        let delegate = AudioControllerTestDelegate(config, context: context)
+        controller.context?.listeners = [delegate]
+        context.stageInstances = [delegate]
+        let processFrameExpectation = expectation(description: "testStartStreaming calls AudioControllerTestDelegate as the result of processFrame method completion")
 
         // AudioControllerDelegate processFrame is called
         XCTAssertNoThrow(try AVAudioSession.sharedInstance().setCategory(.record))
         delegate.asyncExpectation = processFrameExpectation
-        XCTAssertNotNil(controller.delegate)
         XCTAssertNotNil(delegate.asyncExpectation)
```

Comment:
Does this line test anything in your code?

Reply:
it's a bit of defensiveness for tracking down whether a failure is due to test threading or due to a failed test assertion.